packages/next/next-server/lib/router/router.ts

```diff
@@ -617,8 +623,8 @@ export default class Router implements BaseRouter {
           window.location.href = as
 
           // Changing the URL doesn't block executing the current code path.
-          // So, we need to mark it as a cancelled error and stop the routing logic.
-          err.cancelled = true
+          // So, we need to mark it as aborted and stop the routing logic.
+          err.aborted = true
```

Comment:
I'd prefer not to introduce a new property for the user. Can we go back to using `cancelled`?

Reply:
@Timer currently the `.cancelled` property is being hijacked to also stop the routing logic in cases where the route change isn't explicitly cancelled, like when `getServerSideProps` errors or something. This made those errors being ignored in the routing events, since an error event is emitted for cancellation, but these errors are generated outside of that flow. The `.aborted` property was added to distinguish this kind of errors and emit an event for those. I didn't intend for this property to be public, I just needed the `.cancelled` property to be `undefined`. I've changed it to use an internal symbol instead.